Northeastern-Electric-Racing/FinishLine
# #1522: added create material type endpoint with tests

## Changes

- Added `create material type` endpoint in Projects routes, controller, and services.
- Added backend tests for `create material type` endpoint.

## Test Cases

- Case A
- Check the submitter of the endpoint access denied case (not at least in leadership role)
- Case B 
- Check if the material type is already created
- Case C
- Check the endpoint is working as expected

## Screenshots

### Check access denied case when the submitter is MEMBER:
<img width="908" alt="#1522-create-material-type-endpoint-failed" src="https://github.com/Northeastern-Electric-Racing/FinishLine/assets/91857946/02d00ee7-ef86-402d-9a2f-a4f8a09bf113">

### Check HttpException in case the material type is already defined:
<img width="905" alt="image" src="https://github.com/Northeastern-Electric-Racing/FinishLine/assets/91857946/2fbe1885-6e50-46f7-8cc0-fcb77bd8fcbd">

### The endpoint works fine as expected:
<img width="909" alt="#1522-create-material-type-endpoint" src="https://github.com/Northeastern-Electric-Racing/FinishLine/assets/91857946/696f42e4-bb13-4a95-a5cb-e353b7e79042">

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.


Closes # (issue #)
